routes/web.php

```diff
@@ -27,4 +26,7 @@
     Route::get('/dashboard', function () {
         return view('dashboard');
     })->name('dashboard');
+    Route::resource('/blog',PostsController::class)->except(['index','show']);
 });
+
+Route::get('/blog/{slug}',[PostsController::class,'show'])->name('blog.show');
```

Comment:
Since this is a catch all route, I have placed it towards the bottom to avoid conflicts with `/blog/edit`. You can work around this by using route where regex or prefix the post as `/post/{slug}` but this is the most readable solution